pkg/runner/verify_test.go

```
@@ -0,0 +1,51 @@
+	"github.com/projectdiscovery/naabu/v2/pkg/scan"
+	"github.com/stretchr/testify/require"
+)
+
+// TestConnectVerificationBoundedAndVerifies ensures the verification pass keeps
+// verifying open ports correctly after switching to a bounded sizedwaitgroup
+// (so concurrent dials are capped at the scan rate instead of one goroutine per
+// host).
+func TestConnectVerificationBoundedAndVerifies(t *testing.T) {
+	ln, err := net.Listen("tcp", "127.0.0.1:0")
+	require.NoError(t, err)
+	defer ln.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `ln.Close` is not checked

(errcheck)